### ISSUE TITLE
read_params

### ISSUE THREAD

**Issue by dberenbaum** [iterative/dvclive#415]
Add a method to read params from a file. This can make it easy to start using file-based params (encouraged in dvc pipelines and good practice generally) with dvclive. 

Sample docs for this method:

> # Live.read_params()
> 
> ```
>  def read_params(path: str):
> ```
> 
> ## Usage
> 
> ```
> from dvclive import Live
> 
> live = Live()
> 
> params = live.read_params("params.yaml")
> 
> model(**params)
> ```
> 
> ## Description
> 
> Read a YAML, JSON, TOML, or simple Python file into a dict. 
>
> If ~`save_dvc_exp=True`~ `dvcyaml=True`, the path will also be specified as a parameters file in `dvc.yaml`.
> ...

Compared to [`dvc.api.params_show`](https://dvc.org/doc/api-reference/params_show), this method:
1. Doesn't require DVC.
2. Adds the file as a params file instead of needing it to already be setup in `dvc.yaml`.
3. Is more of a simple convenience for reading config files -- there are no stages, revisions, etc.

**Comment by dberenbaum** [comment]
Hopefully this can help with transitioning to pipelines. We could still encourage usage of a separate params dep file even when using `dvc_exp_save` and no pipeline. If they already have a `params.yaml` and use `read_params`, then adding a pipeline can be as simple as `dvc stage add -n train -d params.yaml -o dvclive ... python train.py` (or a pre-filled dvc.yaml with help from vs code).

**Comment by skshetry** [comment]
Do users find it hard to do this?
```python
with open("params.yaml") as f:
    params = yaml.load("params.yaml")

with open("params.toml", "rb") as f:
    params = tomllib.load(f)
```

**Comment by daavoo** [comment]
> Do users find it hard to do this?
> 
> ```python
> with open("params.yaml") as f:
>     params = yaml.load(f)
> 
> with open("params.toml", "rb") as f:
>     params = tomllib.load(f)
> ```

To my understanding, the main value comes from:

> Adds the file as a params file instead of needing it to already be setup in dvc.yaml.

**Comment by dberenbaum** [comment]
Lowering to p2 since it doesn't enable/break any workflow ATM

**Comment by mertbozkir** [comment]
I'm curious what will be difference between already implemented **`_read_params`** function?

and I would love to work on this issue! 👊🏻

    
```py
def _read_params(self):
        if os.path.isfile(self.params_file):
            params = load_yaml(self.params_file)
            self._params.update(params)
```

**Comment by dberenbaum** [comment]
Hi @mertbozkir 👋 ! Good question. There are a few differences: 

1. A public `read_params` method should take a filename as an argument and read files from there. It should also add that file path to the top-level `params` section of the `dvc.yaml` file.
1. `_read_params` adds parameters to `self._params`, which will later dump them back to `dvclive/params.yaml`. For a public `read_params` method, we don't want to do this since it will duplicate the parameters (the user already has them in a file in this case).
2. `_read_params` doesn't return anything. A public `read_params` should return the dictionary of params so the parameters can be used in code.

**Comment by skshetry** [comment]
> A public read_params method should take a filename as an argument and read files from there. It should also add that file path to the top-level params section of the dvc.yaml file.

A query API should not have any side-effects, especially an API that is named `read_`.

**Comment by dberenbaum** [comment]
@skshetry Maybe there's a better name like `load_params`? Adding to `dvc.yaml` won't happen at the same time. It would be saved internally to be written during `make_dvcyaml()`.

**Comment by shcheklein** [comment]
>A public read_params method should take a filename as an argument and read files from there. It should also add that file path to the top-level params section of the dvc.yaml file.

I also got confused by this :). @dberenbaum could you please review / add more color to the initial requirements?

**Comment by dberenbaum** [comment]
The goal is to address these points from the initial requirements at the top of the issue:

> If `save_dvc_exp=True`, the path will also be specified as a parameters file in `dvc.yaml`.

I would update this to "If `dvcyaml=True`, ...`. I don't think the `dvcyaml` kwarg existed when the issue was opened.

> 2\. Adds the file as a params file instead of needing it to already be setup in `dvc.yaml`.

If the file (or some subset of parameters in it) is already specified as params dependencies and top-level params, they get deduped in all products.

The reason I feel it's important for this method is to ensure that any parameters read in by this method in fact show up as parameters for that experiment.

**Comment by shcheklein** [comment]
okay, I see and it makes sense. I would rephrase it a bit. "`Live` tracks all files access by `read_params` via `dvc.yaml` automatically to track the ...". That could reduce the feeling of `A query API should not have any side-effects, especially an API that is named read_.`.
